src/main/java/rx/internal/operators/OnSubscribeFromAsync.java

```diff
@@ -0,0 +1,534 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.internal.operators;
+
+import java.util.Queue;
+import java.util.concurrent.atomic.*;
+
+import rx.*;
+import rx.Observable.OnSubscribe;
+import rx.exceptions.*;
+import rx.functions.Action1;
+import rx.internal.util.RxRingBuffer;
+import rx.internal.util.atomic.SpscUnboundedAtomicArrayQueue;
+import rx.internal.util.unsafe.*;
+import rx.plugins.RxJavaHooks;
+import rx.subscriptions.SerialSubscription;
+
+public final class OnSubscribeFromAsync<T> implements OnSubscribe<T> {
+
+    final Action1<AsyncEmitter<T>> asyncEmitter;
+    
+    final AsyncEmitter.BackpressureMode backpressure;
+    
+    public OnSubscribeFromAsync(Action1<AsyncEmitter<T>> asyncEmitter, AsyncEmitter.BackpressureMode backpressure) {
+        this.asyncEmitter = asyncEmitter;
+        this.backpressure = backpressure;
+    }
+    
+    @Override
+    public void call(Subscriber<? super T> t) {
+        BaseAsyncEmitter<T> emitter;
+        
+        switch (backpressure) {
+        case NONE: {
+            emitter = new NoneAsyncEmitter<T>(t);
+            break;
+        }
+        case ERROR: {
+            emitter = new ErrorAsyncEmitter<T>(t);
+            break;
+        }
+        case DROP: {
+            emitter = new DropAsyncEmitter<T>(t);
+            break;
+        }
+        case LATEST: {
+            emitter = new LatestAsyncEmitter<T>(t);
+            break;
+        }
+        default: {
+            emitter = new BufferAsyncEmitter<T>(t, RxRingBuffer.SIZE);
+            break;
+        }
+        }
+
+        t.add(emitter);
+        t.setProducer(emitter);
+        asyncEmitter.call(emitter);
```

Comment:
Why would we do that? The subscribe() itself triggers the execution of the body that sets up the push source.